internal/compiler/widgets/cosmic/lineedit.slint

```diff
@@ -62,6 +62,7 @@ export component LineEdit {
         background: CosmicPalette.control-background;
         border-width: 1px;
         border-color: CosmicPalette.control-divider;
+        clip: true;
```

Comment:
When combining a reformat of whole files and a bug fix, please use separate commits. As a reviewer it's better if I don't have to guess which changes are reformatting and which changes actually fix the bug. Here I'm guessing... that this is the actual bug fix and there are no other changes in this PR that related to the bug fix?

That aside, I'm curious as to why the clip is needed? Is this perhaps a bug in `TextInput` that it is drawing the cursor outside of its bounds? Or does the `TextInput` have the wrong size?

Reply:
> > When combining a reformat of whole files and a bug fix, please use separate commits. As a reviewer it's better if I don't have to guess which changes are reformatting and which changes actually fix the bug. Here I'm guessing... that this is the actual bug fix and there are no other changes in this PR that related to the bug fix?
> 
> Ok will do it in the future. Is it ok now for this PR?

Sure.

> > That aside, I'm curious as to why the clip is needed? Is this perhaps a bug in TextInput that it is drawing the cursor outside of its bounds? Or does the TextInput have the wrong size?
> 
> The second one.

Wouldn't it be better to fix that instead then?

I can see this:

    `width: max(parent.width - self.text-cursor-width, self.preferred-width);`
    
Why isn't it just `parent.width` like the placeholder?

Reply:
> Why isn't it just parent.width like the placeholder?

To answer my own question here: Because the `LineEditBase` implements the scrolling by shifting the `TextInput` based on the cursor position.

As a consequence, `width: 100%` breaks scrolling. That's visible with the testcase of #6243.